lib/compat/wordpress-6.0/class-wp-theme-json-gutenberg.php

```diff
@@ -675,6 +676,18 @@ public function get_custom_templates() {
 		return $custom_templates;
 	}
 
+	/**
+	 * Returns the current theme's wanted patterns(slugs) to be
+	 * registered from Pattern Directory.
+	 *
+	 * @return array|null
+	 */
+	public function get_pattern_settings() {
+		if ( isset( $this->theme_json['patterns'] ) && is_array( $this->theme_json['patterns'] ) ) {
```

Comment:
Can we return an empty array if the theme doesn't register any pattern?

Also, the other public methods that return data from the `theme.json` follow this pattern `get_<key_in_theme_json>`. What do you think of following that naming and calling this `get_patterns`?

Reply:
I renamed as suggested, but why do you think we should return an `empty array`.

Reply:
The main reason for me would be following what we do in the other public methods that are similar (`get_settings`, `get_custom_templates`, `get_template_parts`). There's also no need for consumers to do any type check if the return type is stable.

Reply:
I updated!